R/volcanoPlot.R

```diff
@@ -114,244 +108,381 @@ volcanoPlot <- function(contrastDF,
         }
 
         if (!is.null(foldChangeLines)) {
-            decorations <- getCxPlotDecorations(decorations = decorations,
-                                                color = symbolFill[which(groupNames == "Increased")],
-                                                width = refLineThickness,
-                                                x     = foldChangeLines)
-            decorations <- getCxPlotDecorations(decorations = decorations,
-                                                color = symbolFill[which(groupNames == "Decreased")],
-                                                width = refLineThickness,
-                                                x     = -1*foldChangeLines)
+            decorations <- .getCxPlotDecorations(decorations = decorations,
+                                                 color       = colors[2],
+                                                 width       = refLineThickness,
+                                                 x           = foldChangeLines)
+            decorations <- .getCxPlotDecorations(decorations = decorations,
+                                                 color       = colors[1],
+                                                 width       = refLineThickness,
+                                                 x           = -foldChangeLines)
         }
 
-        # Assign null if footnote is missing
-        if (missing(footnote)) {
-            footnote <- NULL
+        events <- htmlwidgets::JS("{ 'mousemove' : function(o, e, t) {
+                                                if (o != null && o != false) {
+                                                  if (o.y != null &&
+                                                      o.y.data != null &&
+                                                      o.y.smps != null) {
+                                                      info = '<b>' + o.y.vars[0]  + '</b>' + '<br/>' +
+                                                             '<i>' + o.z.Group  + '</i><br/>' +
+                                                             '<b>' + o.y.smps[0]  + '</b>' + ': ' + o.y.data[0][0] + '<br/>' +
+                                                             '<b>' + o.y.smps[1]  + '</b>' + ': ' + o.y.data[0][1] ;
+                                                      if (o.z != null && o.z['GeneName'] != null) {
+                                                        info  = info + '<br/>' +
+                                                              '<b> Symbol</b>' + ': ' + o.z['GeneName'] ;
+                                                      }
+                                                    t.showInfoSpan(e, info);
+                                                  }
+                                                }; }}")
+
+        cx.data <- contrastDF %>% dplyr::select(c(logRatioCol, negLog10P))
+
+        if (missing(geneSymCol) && sizeByIntensity) {
```

Comment:
I thought we were removing these 4 if statements? Separate out the conditional statements for geneSymCol and sizeByIntensity